src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -71,12 +75,43 @@ public ActiveSolutionBoundTracker(IHost host, IActiveSolutionTracker activeSolut
             this.ProjectKey = this.solutionBindingInformationProvider.GetProjectKey();
         }
 
+
         private void OnActiveSolutionChanged(object sender, EventArgs e)
         {
+            UpdateConnection();
+
             this.RaiseAnalyzersChangedIfBindingChanged();
             this.errorListInfoBarController.Refresh();
         }
 
+        private void UpdateConnection()
+        {
+            ISonarQubeService sqService = this.extensionHost.SonarQubeService;
+            if (sqService.IsConnected)
+            {
+                sqService.Disconnect();
+
+                if (this.extensionHost.ActiveSection?.DisconnectCommand.CanExecute(null) == true)
```

Comment:
Tell the TE plugin to clear its connection information.
This in turn will trigger a _BindingStateChange_ event. However, we can tell that nothing has changed and do nothing in that case.

Reply:
TODO: perhaps we should disconnect TE section first so that the WPF commands are removed before we close the server connection.